include/benchmark/benchmark_api.h

```diff
@@ -227,6 +231,126 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 // FIXME Add ClobberMemory() for non-gnu compilers
 #endif
 
+
+
+class Counter {
+public:
+
+  typedef enum {
```

Comment:
And let's make this a non-bitmask enum so we can store `Flags` instead of `uint32_t`.  I would add a `kAvgThreadRate` to handle that case.